[Bug]: Gemini with OpenAI endpoint panics when streaming

### Prerequisites

- [x] I have searched existing issues and discussions to avoid duplicates
- [x] I am using the latest version (or have tested against main/nightly)

### Description

Bifrost crashes when using Gemini through the OpenAI responses endpoint with streaming. I'm using `maximhq/bifrost:v1.3.13`, although it also happens in `maximhq/bifrost:v1.3.12`

### Steps to reproduce

1. Start `bifrost` with `docker run -p 8080:8080 -v $(pwd)/data:/app/data maximhq/bifrost maximhq/bifrost:v1.3.13`
2. Go to `http://localhost:8080` and add an API Key for Gemini
3. Make the following request:
```
curl --location 'http://localhost:8080/openai/responses' \
--header 'Content-Type: application/json' \
--data '{
    "model": "gemini/gemini-2.5-flash",
    "stream": true,
    "input": [
       { "role": "user", "content": "Hello!" }
    ]
  }'
```

`bifrost` will crash with the following error:

```
{"level":"info","time":"2025-10-30T19:00:42Z","message":"successfully started bifrost, serving UI on http://0.0.0.0:8080"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x12e0e40]

goroutine 2868 [running]:
github.com/maximhq/bifrost/transports/bifrost-http/integrations.(*GenericRouter).handleStreamingRequest.(*GenericRouter).handleStreaming.func1(0x400030a400)
        github.com/maximhq/bifrost/transports/bifrost-http/integrations/utils.go:726 +0x200
github.com/valyala/fasthttp.NewStreamReader.func1()
        github.com/valyala/fasthttp@v1.65.0/stream.go:44 +0x40
created by github.com/valyala/fasthttp.NewStreamReader in goroutine 1805
        github.com/valyala/fasthttp@v1.65.0/stream.go:43 +0x284
```

### Expected behavior

bifrost should get a response without crashing

### Actual behavior

bifrost panics when streaming the response, works fine without streaming

### Affected area(s)

Core (Go)

### Version

v1.3.13,v1.3.12

### Environment

```text
- MacOS 15.6.1
- Docker version 28.3.3
```

### Relevant logs/output

```shell

```

### Regression?

_No response_

### Severity

High (major functionality broken)

--- a/core/schemas/mux.go
+++ b/core/schemas/mux.go
@@ -1057,6 +1057,8 @@ func (cr *BifrostChatResponse) ToBifrostResponsesStreamResponse() *BifrostRespon
 		}
 	}
 
+	streamResp.ExtraFields.RequestType = ResponsesStreamRequest
+
 	// Return the created stream response
 	return streamResp
 }

--- a/transports/bifrost-http/integrations/utils.go
+++ b/transports/bifrost-http/integrations/utils.go
@@ -722,10 +722,10 @@ func (g *GenericRouter) handleStreaming(ctx *fasthttp.RequestCtx, config RouteCo
 				switch {
 				case chunk.BifrostTextCompletionResponse != nil:
 					convertedResponse, err = config.StreamConfig.TextStreamResponseConverter(chunk.BifrostTextCompletionResponse)
-				case chunk.BifrostChatResponse != nil:
-					convertedResponse, err = config.StreamConfig.ChatStreamResponseConverter(chunk.BifrostChatResponse)
 				case chunk.BifrostResponsesStreamResponse != nil:
 					convertedResponse, err = config.StreamConfig.ResponsesStreamResponseConverter(chunk.BifrostResponsesStreamResponse)
+				case chunk.BifrostChatResponse != nil:
+					convertedResponse, err = config.StreamConfig.ChatStreamResponseConverter(chunk.BifrostChatResponse)
 				case chunk.BifrostSpeechStreamResponse != nil:
 					convertedResponse, err = config.StreamConfig.SpeechStreamResponseConverter(chunk.BifrostSpeechStreamResponse)
 				case chunk.BifrostTranscriptionStreamResponse != nil:

